contracts/cw-core/src/tests.rs

```diff
@@ -821,9 +829,164 @@ fn do_standard_instantiate(auto_add: bool) -> (Addr, App) {
     (gov_addr, app)
 }
 
+#[test]
+fn test_admin_permissions() {
+    let (core_addr, mut app) = do_standard_instantiate(true, None);
+
+    let start_height = app.block_info().height;
+    let proposal_modules: Vec<Addr> = app
+        .wrap()
+        .query_wasm_smart(
+            core_addr.clone(),
+            &QueryMsg::ProposalModules {
+                start_at: None,
+                limit: None,
+            },
+        )
+        .unwrap();
+
+    assert_eq!(proposal_modules.len(), 1);
+    let proposal_module = proposal_modules.into_iter().next().unwrap();
+
+    // Random address can't call ExecuteAdminMsgs
+    let res = app.execute_contract(
+        Addr::unchecked("random"),
+        core_addr.clone(),
+        &ExecuteMsg::ExecuteAdminMsgs {
+            msgs: vec![WasmMsg::Execute {
+                contract_addr: core_addr.to_string(),
+                msg: to_binary(&ExecuteMsg::Pause {
+                    duration: Duration::Height(10),
+                })
+                .unwrap(),
+                funds: vec![],
+            }
+            .into()],
+        },
+        &[],
+    );
+    assert!(res.is_err());
+
+    // Proposal mdoule can't call ExecuteAdminMsgs
+    let res = app.execute_contract(
+        proposal_module.clone(),
+        core_addr.clone(),
+        &ExecuteMsg::ExecuteAdminMsgs {
+            msgs: vec![WasmMsg::Execute {
+                contract_addr: core_addr.to_string(),
+                msg: to_binary(&ExecuteMsg::Pause {
+                    duration: Duration::Height(10),
+                })
+                .unwrap(),
+                funds: vec![],
+            }
+            .into()],
+        },
+        &[],
+    );
+    assert!(res.is_err());
+
+    // Update Admin can't be called by non-admins
+    let res = app.execute_contract(
+        Addr::unchecked("rando"),
+        core_addr.clone(),
+        &ExecuteMsg::UpdateAdmin {
+            admin: Some(Addr::unchecked("rando")),
+        },
+        &[],
+    );
+    assert!(res.is_err());
+
+    // Update Admin can't be called, even by the proposal module
+    let res = app.execute_contract(
+        proposal_module.clone(),
```

Comment:
Think we already have a test for this above. How about checking that the DAO can't call this on itself? This is how a proposal module's message would get executed.
```suggestion
    // Update Admin can't be called, even by the core module
    let res = app.execute_contract(
       core_addr.clone(),
```